docs/_docs/reference/contextual/derivation.md

```diff
@@ -26,30 +26,48 @@ given [T: Show]     : Show[Tree[T]]     = Show.derived
 
 We say that `Tree` is the _deriving type_ and that the `Eq`, `Ordering` and `Show` instances are _derived instances_.
 
-### Types supporting `derives` clauses
+More formally, when a class/trait/object/enum `DerivingType` `derives` a type class `TC`, the following given instance is created in `DerivingType`'s companion object (or `DerivingType` itself if it is an object):
+* if `DerivingType` doesn't have type parameters
+```scala
+given TC[DerivingType] = TC.derived
+```
+* if `DerivingType` has type parameters `[T_1, ... T_N]`
```

Comment:
These rules are further refined in https://github.com/lampepfl/dotty/blob/main/compiler/src/dotty/tools/dotc/typer/Deriving.scala

Reply:
mostly for higher kinded types